iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/services/Devices.java

```diff
@@ -218,9 +218,19 @@ public DeviceTwinName getDeviceTwinNames()
                         } else {
                             this.deviceTwinClient.updateTwin(device.getTwin().toDeviceTwinDevice());
                             DevicePropertyServiceModel model = new DevicePropertyServiceModel();
-                            model.setTags(new HashSet<String>(device.getTwin().getTags().keySet()));
+                            model.setTags(new HashSet<String>(
```

Comment:
these would be a lot more readable with an if statement instead of ternary. Ex
if(device.getTwin() != null && device.getTwin().getTags() != null) {
    model.setTags(...)
}
and similarly for the other ones.

Reply:
+1

Reply:
+1

Reply:
done